From: Thomas Renninger <trenn@suse.de>
Subject: kernel/module.c: Fix compiler warnings if debug is compiled in
References: none
Patch-Mainline: queued for 2.6.37 (in a Greg tree, no idea which)


kernel/module.c: Fix compiler warnings if debug is compiled in

Fixes these warnings:
kernel/module.c: In function ‘layout_sections’:
kernel/module.c:1776: error: ‘name’ undeclared (first use in this function)
kernel/module.c:1776: error: (Each undeclared identifier is reported only once
kernel/module.c:1776: error: for each function it appears in.)
kernel/module.c: In function ‘move_module’:
kernel/module.c:2394: warning: format ‘%lx’ expects type ‘long unsigned int’,
but argument 2 has type ‘Elf64_Addr’

if DEBUGP is defined in kernel/module.c

Signed-off-by: Thomas Renninger <trenn@suse.de>
CC: greg@kroah.com
CC: jbaron@redhat.com


---
 kernel/module.c |    5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

Index: linux-2.6.35-master/kernel/module.c
===================================================================
--- linux-2.6.35-master.orig/kernel/module.c
+++ linux-2.6.35-master/kernel/module.c
@@ -1846,7 +1846,6 @@ static void layout_sections(struct modul
 			    || strstarts(sname, ".init"))
 				continue;
 			s->sh_entsize = get_offset(mod, &mod->core_size, s, i);
-			DEBUGP("\t%s\n", name);
 		}
 		if (m == 0)
 			mod->core_text_size = mod->core_size;
@@ -2463,8 +2462,8 @@ static int move_module(struct module *mo
 			memcpy(dest, (void *)shdr->sh_addr, shdr->sh_size);
 		/* Update sh_addr to point to copy in image. */
 		shdr->sh_addr = (unsigned long)dest;
-		DEBUGP("\t0x%lx %s\n",
-		       shdr->sh_addr, info->secstrings + shdr->sh_name);
+		DEBUGP("\t0x%p %s\n",
+		       (void *)shdr->sh_addr, info->secstrings + shdr->sh_name);
 	}
 
 	return 0;
